src/IK/src/InverseKinematics.cpp

```diff
@@ -0,0 +1,140 @@
+#include <BiomechanicalAnalysis/IK/InverseKinematics.h>
+#include <iostream>
+
+using namespace BiomechanicalAnalysis::IK;
+
+
+bool HumanIK::initialize(std::weak_ptr<const BipedalLocomotion::ParametersHandler::IParametersHandler> handler,
+                std::shared_ptr<iDynTree::KinDynComputations> kinDyn)
+{
+    constexpr std::size_t highPriority = 0;
+    constexpr std::size_t lowPriority = 1;
+
+    m_jointPositions.resize(kinDyn->getNrOfDegreesOfFreedom());
+    m_jointVelocities.resize(kinDyn->getNrOfDegreesOfFreedom());
+
+    std::cout << "nr of dofs = " << kinDyn->getNrOfDegreesOfFreedom() << std::endl;
+
+    auto ptr = handler.lock();
+    if (ptr == nullptr)
+    {
+        std::cerr << "[HumanIK::initialize] Invalid parameter handler." << std::endl;
+        return false;
+    }
+
+    bool ok = m_qpIK.initialize(ptr->getGroup("IK"));
+    auto group = ptr->getGroup("IK").lock();
+    std::string variable;
+    group->getParameter("robot_velocity_variable_name", variable);
+    m_variableHandler.addVariable(variable, kinDyn->getNrOfDegreesOfFreedom() + 6);
+
+    m_link1OrientationTask = std::make_shared<BipedalLocomotion::IK::SO3Task>();
+    ok = ok && m_link1OrientationTask->setKinDyn(kinDyn);
+    ok = ok && m_link1OrientationTask->initialize(ptr->getGroup("LINK1_TASK"));
+    ok = ok && m_qpIK.addTask(m_link1OrientationTask, "link1_task", highPriority);
+
+    m_qpIK.finalize(m_variableHandler);
+
+    return true;
+}
+
+bool HumanIK::setDt(const double dt)
+{
+    m_dtIntegration = dt;
+
+    return true;
+}
+
+double HumanIK::getDt() const
+{
+    return m_dtIntegration;
+}
+
+bool HumanIK::setDoFsNumber(const int nrDoFs)
+{
+    m_nrDoFs = nrDoFs;
+    m_jointPositions.resize(m_nrDoFs);
+    m_jointVelocities.resize(m_nrDoFs);
+
+    return true;
+}
+
+int HumanIK::getDoFsNumber() const
+{
+    return m_nrDoFs;
+}
+
+bool HumanIK::setInitialJointPositions(const Eigen::Ref<const Eigen::VectorXd> qInitial)
+{
+    m_jointPositions = qInitial;
+
+    return true;
+}
+
+bool HumanIK::setLink1OrientationAndAngVel(const manif::SO3d &link1Orientation,
+                                           const manif::SO3Tangentd &link1AngularVelocity)
+{
+    m_link1OrientationTask->setSetPoint(link1Orientation, link1AngularVelocity);
+
+    return true;
+}
+
+bool HumanIK::advance()
+{
+    bool ok{true};
+    ok = ok && m_qpIK.advance();
+    ok = ok && m_qpIK.isOutputValid();
+
+    if(ok)
+    {
+        m_jointVelocities = m_qpIK.getOutput().jointVelocity;
+        m_baseVelocity = m_qpIK.getOutput().baseVelocity;
+    }
+
+    // integrate the joint velocities
+    m_jointPositions += m_dtIntegration * m_jointVelocities;
```

Comment:
Here there is no integration step for the base. As discussed we could exploit the integrator class from blf (https://github.com/ami-iit/bipedal-locomotion-framework/blob/a3a8e9cb8a0c3532db81d814d4851009f8134195/src/ContinuousDynamicalSystem/include/BipedalLocomotion/ContinuousDynamicalSystem/Integrator.h#L41)

Reply:
It also seems to me that this joint position is not used to update the kindyn object. In the current implemntation you think this should be done internally to this class or externally?

Reply:
> In the current implemntation you think this should be done internally to this class or externally?

In the current implementation it has to be done externally, but I can change it and do it in this method.

Reply:
> As discussed we could exploit the integrator class from blf

Yes, I will add it.

Reply:
I added an integrator taking inspiration from here:
https://github.com/ami-iit/bipedal-locomotion-framework/blob/026c79fffb81b1c5f4a9239bfdb4f025de95bbd7/src/IK/tests/QPInverseKinematicsTest.cpp#L410-L431